connector-pagerduty/src/main/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyTransformer.java

```diff
@@ -45,30 +48,43 @@ public class PagerDutyTransformer implements Processor {
     @Override
     public void process(Exchange exchange) {
         JsonObject cloudEventPayload = exchange.getIn().getBody(JsonObject.class);
+        validatePayload(cloudEventPayload);
 
         JsonObject message = new JsonObject();
         message.put(EVENT_ACTION, PagerDutyEventAction.TRIGGER);
         message.mergeIn(getClientLink(cloudEventPayload, cloudEventPayload.getString(ENVIRONMENT_URL)));
 
         JsonObject messagePayload = new JsonObject();
         messagePayload.put(SUMMARY, cloudEventPayload.getString(EVENT_TYPE));
-        messagePayload.put(TIMESTAMP, LocalDateTime.parse(cloudEventPayload.getString(TIMESTAMP)).format(PD_DATE_TIME_FORMATTER));
+
+        String timestamp = cloudEventPayload.getString(TIMESTAMP);
+        if (timestamp != null) {
+            try {
+                messagePayload.put(TIMESTAMP, LocalDateTime.parse(timestamp).format(PD_DATE_TIME_FORMATTER));
+            } catch (DateTimeParseException e) {
+                Log.warnf(e, "Unable to parse timestamp %s, dropped from payload", timestamp);
+            } catch (DateTimeException e) {
+                Log.warnf(e, "Timestamp %s was successfully parsed, but could not be reformatted for PagerDuty, dropped from payload", timestamp);
+            }
+        } else {
+            Log.warn("Timestamp not provided for payload, ignored");
+        }
+
```

Comment:
It seems like the `BaseTransfomer` class will always add the `Timestamp` field of the action. Have you found any edge case where we might not be adding the timestamp? Because in that case we should fix it for all the processors in the engine :eyes:

Reply:
Whoops, my bad. Misread the code :facepalm:

Reply:
Actually not, I think we can spare the null check for the timestamp value. Sorry for the confusion :sweat:

Reply:
No it should be safe to take them out, I'll update this.